packages/core-data/src/queried-data/actions.js

```diff
@@ -17,6 +17,25 @@ export function receiveItems( items ) {
 	};
 }
 
+/**
+ * Returns an action object used in signalling that entity records have been
+ * deleted and it needs to be removed from entities state.
+ *
+ * @param {string} kind            Kind of the removed entity.
+ * @param {string} name            Name of the removed entity.
+ * @param {Array|Object} records   Records removed.
+ * @return {Object} Action object.
+ */
+export function removeItems( kind, name, records ) {
+	return {
+		type: 'REMOVE_ITEMS',
+		items: castArray( records ),
+		kind,
+		name,
+		invalidateCache: false,
```

Comment:
This needs to be a parameter, only set to true in `deleteEntityRecord`, and handled in `getEntityRecords.shouldInvalidate`.

Reply:
Why is this needed? How does `getEntityRecords.shouldInvalidate` work and how does `deleteEntityRecord` impact that? Currently I am not treating this action in `getEntityRecords.shouldInvalidate` and it works. When does it break?

This line is momentarily cruft :)